src/composition-engine.js

```diff
@@ -85,18 +116,26 @@ export class CompositionEngine {
 
   _createControllerAndSwap(context) {
     function swap(controller) {
-      return Promise.resolve(context.viewSlot.removeAll(true)).then(() => {
-        if (context.currentController) {
-          context.currentController.unbind();
-        }
-
-        context.viewSlot.add(controller.view);
-
-        if (context.compositionTransactionNotifier) {
-          context.compositionTransactionNotifier.done();
-        }
-
-        return controller;
+      let swapStrategy;
+
+      swapStrategy = context.swapOrder in swapStrategies
+              ? swapStrategies[context.swapOrder]
+              : swapStrategies.after;
+
+      swapStrategy(context.viewSlot, context.viewSlot.children[0], () => {
+        return Promise.resolve().then(() => {
```

Comment:
Is there a reason to lose a turn with an empty `Promise.resolve()`? I don't see a reason to.

If yes it should be commented.
If not, you're probably better off with: 
`Promise.resolve(context.viewSlot.add(controller.view)).then()`.